### DISCUSSION TITLE
How were the maximum safe frequenies determined?

### DISCUSSION THREAD

**Discussion by M1cha** [hanai3Bi/Switch-OC-Suite#36] [Q&A] [answered]
Were they tested or were they found in a datasheet?

**Comment by hanai3Bi** [comment, marked as answer]
they were tested by switch expert developers and i just copy and paste.
for example, 998 GPU is the limit that doesn't exceed gpu pmic current limit even in worst binned consoles(without undervolt)

for dram, when Kazu was developing OCS he put it as safe freq because it's freq that worked for most consoles, but dram oc capabilities differ for each console

**Comment by hanai3Bi** [comment]
you can consider what l4t allows as safe values, but keep in mind that it doesn't mean you can max out both CPU and GPU.
and also l4t monitors values and throttles unlike HOS which doesn't have such safety measures

**Comment by M1cha** [comment]
Okay that makes sense. What about RAM(only) OC? (which reportly seems to improve TOTK performance a lot). Is there a technical reason that on mariko 1600(stock) and 1996 (max safe) are the only options with no steps in between?

**Reply by hanai3Bi** [reply]
for Mariko, Nvidia put only 3 tables for dram: 204, 1331, 1600. 204 is used for sleep, so there are only 2 available 
so 1331 is replaced with 1600 and 1600 is replaced with OC one in OC Suite
